src/meta/rb_map.lean

```diff
@@ -177,9 +186,26 @@ l.foldr (λ n s', s'.insert n) s
 
 end name_set
 
+/-! ### Declarations about `name_map` -/
+
 namespace name_map
 
 meta instance {data : Type} : inhabited (name_map data) :=
 ⟨mk_name_map⟩
 
 end name_map
+
+/-! ### Declarations about `expr_map` -/
+
+namespace expr_map
+
+/--
+`find_defeq red m e` looks for a key in `m` that is defeq to `e` (up to transparency `red`),
+and returns the value associated with this key if it exists.
+Otherwise, it fails.
+-/
+meta def find_defeq (red : tactic.transparency) {v} (m : expr_map v) (e : expr) :
```

Comment:
`expr_map` seems to be very much the wrong data structure here.  It's probably better to refactor this to use lists (since the only application I could find doesn't do rb_map lookups).

Reply:
Good call. The `expr_map` made sense back when we were matching atoms with syntactic equality.